.github/scripts/workflow-monitor.py

```diff
@@ -32,44 +35,71 @@
 STOP_STATES = []
 NOP_STATES = ["action_required"] # TODO: unsure when this happens
 
-def main(platform: Platform, workflow_id: str, gha_ci_personal_token: str):
+def wrap_in_code(wrap: str)
+    return f"\n```\n{wrap}\n```"
 
+def main(platform: Platform):
     consecutive_failures = 0
-    headers = {'Authorization': "token {}".format(gha_ci_personal_token.strip())}
-    gha_workflow_api_url = f"{gha_runs_api_url}/{workflow_id}"
 
+    print("Workflow monitor started")
     platform_lib = get_platform_lib(platform)
-    while True:
-        time.sleep(POLLING_INTERVAL_SECONDS)
-
-        res = requests.get(gha_workflow_api_url, headers=headers)
-        if res.status_code == 200:
-            consecutive_failures = 0
-            res_dict = res.json()
-            state_status = res_dict["status"]
-            state_concl = res_dict["conclusion"]
-
-            print("Workflow {} status: {} {}".format(workflow_id, state_status, state_concl))
-            if state_status in ['completed']:
-                if state_concl in TERMINATE_STATES:
-                    platform_lib.terminate_instances(gha_ci_personal_token, workflow_id)
-                    exit(0)
-                elif state_concl in STOP_STATES:
-                    platform_lib.stop_instances(gha_ci_personal_token, workflow_id)
-                    exit(0)
-                elif state_concl not in NOP_STATES:
-                    print("Unexpected Workflow State On Completed: {}".format(state_concl))
+
+    try:
+        print("Sending startup message")
+        issue_post(ci_env['PERSONAL_ACCESS_TOKEN'], f"Workflow monitor started for CI run: {ci_env['GITHUB_RUN_ID']}")
```

Comment:
Thoughts on not doing this (every commit added will send this message polluting the comment stream).

Reply:
I think it'd be fine just to comment on the issue in the case of a failure of the main loop. I think the checks you have in setup-workflow-monitor.py are sufficient to make sure it's up.

Reply:
Done